pennylane/templates/repeat.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``repeat`` template constructor.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+
+###################
+
+
+@template
+def repeat(circuit, depth, *args, **kwargs):
+    r"""Repeatedly applies a function containing quantum gates or templates.
+
+    Args:
+        circuit (function): A function that applies the quantum gates/templates being repeated.
+        depth (int): The number of time ``circuit`` is repeatedly applied.
+        *args: Dynamic parameters that are passed into ``circuit`` each time it is
+               repeated (see UsageDetails for more information).
+        **kwargs: Static parameters that are passed into ``circuit`` each time it is
+                  repeated (see UsageDetails for more information).
+
+    Raises:
+        ValueError: if inputs do not have the correct format
+
+    .. UsageDetails::
+
+        **Repeating Gates**
+
+        To repeatedly apply a collection of gates/templates, a function applying each of these
+        operations must first be defined. For example, we can define the following circuit:
+
+        .. code-block:: python3
+
+            import pennylane as qml
+            import numpy as np
+
+            def circuit():
+                qml.Hadamard(wires=[0])
+                qml.CNOT(wires=[0, 1])
+                qml.PauliX(wires=[1])
+
+        and then pass it into the ``qml.repeat`` function (in this instance, we repeat ``circuit`` 3 times):
+
+        .. code-block:: python3
+
+            dev = qml.device('default.qubit', wires=3)
+
+            @qml.qnode(dev)
+            def ansatz():
+                qml.repeat(circuit, 3)
+                return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+
+        This creates the following circuit:
+
+        >>> ansatz()
+        >>> print(ansatz.draw())
+        0: ──H──╭C──X──H──╭C──X──H──╭C──X──┤ ⟨Z⟩
+        1: ─────╰X────────╰X────────╰X─────┤ ⟨Z⟩
+
+        **Static Arguments**
+
+        Static arguments are arguments passed into ``circuit`` that *don't change with each
+        repetition of the circuit*. Static parameters are always passed as keyword arguments into ``qml.repeat``.
```

Comment:
```suggestion
        Static arguments are arguments passed into ``circuit`` that don't change with each
        repetition of the circuit. Static parameters are always passed as keyword arguments into ``qml.repeat``.
```
Italics are a bit dramatic. I think this reads better without it